packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,336 @@
+import 'dart:convert';
+
+import 'package:flutter_tools/src/base/process.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+import 'package:vm_snapshot_analysis/treemap.dart';
+
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+// This file can be in devtools_shared
+class SizeAnalyzer {
+  static const String kAotSizeJson = 'aot-size.json';
+
+  static String getAotSizeAnalysisJsonFile(String aotOutputPath) {
+    return globals.fs.path.join(aotOutputPath, kAotSizeJson);
+  }
+
+  static String getAotSizeAnalysisExtraGenSnapshotOption(String aotOutputPath) {
+    return '--print-instructions-sizes-to=${getAotSizeAnalysisJsonFile(aotOutputPath)}';
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  final RegExp _kGetUnzipOutput =
+      RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  Map<String, dynamic> analyzeApkSize(
+      {@required File apk, @required File aotSizeJson}) {
+    globals.printStatus('▒' * 80);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      apk.lengthSync() ~/ 1024,
+      0,
+    );
+    globals.printStatus('━' * 80);
+    final Directory tempApkContent = globals.fs.systemTempDirectory.createTempSync('flutter_tools.');
```

Comment:
why a dot at the end of this temp directory name?

Reply:
I don't love the clarify of this API but apparently it appends a random number after the period as it creates a temp directory.

https://cs.opensource.google/search?q=createTempSync&ss=flutter